Sources/Datadog/Core/Upload/DataUploadDelay.swift

```diff
@@ -6,37 +6,38 @@
 
 import Foundation
 
+internal protocol Delay {
+    func nextUploadDelay() -> TimeInterval
+    mutating func decrease()
+    mutating func increase()
+}
```

Comment:
nit: the name, `Delay` is domain-agnostic, but the `nextUploadDelay()` has "upload" meaning. This
```swift
func next() -> TimeInterval
mutating func decrease()
mutating func increase()
```
would be more coherent IMO.

Reply:
i didn't pay attention before but actually `current` makes more sense than `next`, wdyt?